src/webapp/reports/autogenerated-forms/GridFormViewModel.ts

```diff
@@ -139,33 +139,40 @@ export class GridViewModel {
                 .groupBy(row => row.name.replace(/\s*\(\d+\)$/, ""))
                 .size() === 1;
 
-        const totals = _(columns)
-            .map(column => {
-                const allDataElements = subsections.flatMap(subSection => subSection.dataElements);
-                const selectedDataElements = allDataElements.filter(dataElement =>
-                    section.totals?.dataElementsCodes.includes(dataElement.code)
-                );
-
-                const columnWithDataElements = _(selectedDataElements)
-                    .map((dataElement): Maybe<TotalItem> => {
-                        if (dataElement.type !== "NUMBER") return undefined;
-                        const categoryOptionCombo = dataElement.categoryOptionCombos[0];
-                        if (!categoryOptionCombo) {
-                            console.warn(
-                                `Cannot found categoryOptionCombo in column ${column.name} for dataElement ${dataElement.code}`
-                            );
-                            return undefined;
-                        }
-
-                        return { dataElement, categoryOptionCombo };
-                    })
-                    .compact()
-                    .value();
+        const summary = _(section.totals)
+            .map((sectionTotal, key) => {
+                const cellTotals = columns.map(column => {
+                    const allDataElements = subsections.flatMap(subSection => subSection.dataElements);
+                    const selectedDataElements = allDataElements.filter(dataElement =>
+                        sectionTotal.dataElementsCodes.includes(dataElement.code)
+                    );
+
+                    const columnWithDataElements = _(selectedDataElements)
```

Comment:
I saw in [this PR](https://github.com/EyeSeeTea/d2-autogen-forms/pull/69/files#diff-66095557ec1c63a12af1b90002aef416eaa773b16b192cd3d6be0dce2cca67cdR150) you did a refactor. So probably you'll need to fix the conflict.